### PR TITLE
Get workflow instance by Id & List workflow instances

### PR DESCRIPTION
**Pull request by jackschofield23**
for #62  and #56

### CODE REVIEW

File: src/WorkflowManager/Controllers/WorkflowInstanceController.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            var workflowsInstances = await _workflowInstanceService.GetListAsync();
+
+            return Ok(workflowsInstances);
+        }
+        catch (Exception e)
+        {
+            this._logger.LogError($"{nameof(GetListAsync)} - Failed to get workflowInstances", e);
+
+            return Problem($"Unexpected error occured: {e.Message}", $"/workflowinstances", 500);
```
Comment (lillie-dae) [inline src/WorkflowManager/Controllers/WorkflowInstanceController.cs:52]: possibly use httpstatuscode enum

File: src/WorkflowManager/Controllers/WorkflowInstanceController.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    /// <returns>A list of workflow instances.</returns>
+    [Route("{id}")]
+    [HttpGet]
+    public async Task<IActionResult> GetByIdAsync([FromRoute] string id)
+    {
+        if (string.IsNullOrWhiteSpace(id) || !Guid.TryParse(id, out _))
+        {
+            this._logger.LogDebug($"{nameof(GetByIdAsync)} - Failed to validate {nameof(id)}");
+
+            return Problem($"Failed to validate {nameof(id)}, not a valid guid", $"/workflows/{id}", 400);
```
Comment (lillie-dae) [inline src/WorkflowManager/Controllers/WorkflowInstanceController.cs:69]: see above

File: src/WorkflowManager/Controllers/WorkflowInstanceController.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                this._logger.LogDebug($"{nameof(GetByIdAsync)} - Failed to find workflow instance with Id: {id}");
+
+                return NotFound($"Faild to find workflow instance with Id: {id}");
+            }
+
+            return Ok(workflowInstance);
+        }
+        catch (Exception e)
+        {
+            return Problem($"Unexpected error occured: {e.Message}", $"/workflowinstances/{nameof(id)}", 500);
```
Comment (lillie-dae) [inline src/WorkflowManager/Controllers/WorkflowInstanceController.cs:87]: see above

File: tests/UnitTests/WorkflowManager.Tests/Controllers/WorkflowInstanceControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        [Fact]
+        public async Task GetByIdAsync_ServiceException_ReturnProblem()
+        {
+            var workflowId = Guid.NewGuid().ToString();
+            _workflowInstanceService.Setup(w => w.GetByIdAsync(workflowId)).ThrowsAsync(new Exception());
+
+            var result = await WorkflowInstanceController.GetByIdAsync(workflowId);
+
+            var objectResult = Assert.IsType<ObjectResult>(result);
+            Assert.Equal(500, objectResult.StatusCode);
```
Comment (lillie-dae) [inline tests/UnitTests/WorkflowManager.Tests/Controllers/WorkflowInstanceControllerTests.cs:140]: possibly use status code enum here aswell

**Review by lillie-dae — APPROVED** [review]

File: src/WorkflowManager/Controllers/WorkflowInstanceController.cs (hunk 1 above)
Comment (jackschofield23) [reply src/WorkflowManager/Controllers/WorkflowInstanceController.cs:52]: fixed

File: tests/UnitTests/WorkflowManager.Tests/Controllers/WorkflowInstanceControllerTests.cs (hunk 4 above)
Comment (jackschofield23) [reply tests/UnitTests/WorkflowManager.Tests/Controllers/WorkflowInstanceControllerTests.cs]: fixed

**Pull request merged**